pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -454,6 +454,12 @@ type PDSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum:="";"v1"
 	StartUpScriptVersion string `json:"startUpScriptVersion,omitempty"`
+	// Containers allows injecting additional containers. This is meant to
+	// allow adding an additional container to an PD pod.
+	// Containers described here modify an operator generated container if they
+	// share the same name and modifications are done via a strategic merge
+	// patch.
+	Containers []corev1.Container `json:"containers,omitempty"`
```

Comment:
There is already a field `AdditionalContainers` in the `ComponentSpec`, I think we can just reuse it.

Reply:
BTW, the container name of the components are set by the TiDB Operator, so please help add descriptions in the above field so the user can know how to add new containers or patch existing containers.

Reply:
ok